source/frontend/src/features/leases/detail/LeasePages/payment/modal/period/PeriodForm.tsx

```diff
@@ -64,6 +69,27 @@ export const PeriodForm: React.FunctionComponent<React.PropsWithChildren<IPeriod
     },
   ];
 
+  const initialGstAmount = initialValues.gstAmount;
+
+  const onGstCheckChange = (
+    formikState: FormikProps<FormLeasePeriod>,
+    field: string,
+    values: boolean,
+  ) => {
+    if (values === true) {
+      const gstDecimal = gstConstant !== undefined ? parseFloat(gstConstant.value) : undefined;
+      const calculated = round((formikState.values.paymentAmount as number) * (gstDecimal / 100));
+      formikState.setFieldValue('gstAmount', calculated);
+    } else {
```

Comment:
There is a code path here where **gstDecimal** is undefined (when the GST constant is not defined).
I'm not sure what would be the result of line 81 would be in this case; ie: `calculated = round(... * (undefined/100))`

Would it make sense to put a check here for that case?

Reply:
fair point. I added a better default value